pkg/rpm/resolver/base.go

```diff
@@ -14,109 +14,129 @@ import (
 )
 
 const (
-	prepareBaseScriptName = "prepare-base.sh"
-	prepareBaseScriptLog  = "prepare-base.log"
-	baseImageArchiveName  = "sle-micro-base.tar.gz"
-	baseImageRef          = "slemicro"
+	prepareTarballScriptName = "prepare-tarball.sh"
+	prepareTarballScriptLog  = "prepare-tarball.log"
+	tarballName              = "tarball-image.tar.gz"
+	tarballImgRef            = "tarball"
 )
 
-//go:embed templates/prepare-base.sh.tpl
-var prepareBaseTemplate string
-
-func (r *Resolver) buildBase() error {
-	zap.L().Info("Building base resolver image...")
+//go:embed templates/prepare-tarball.sh.tpl
+var prepareTraballTemplate string
+
+type TarballImageBuilder struct {
+	// dir from where the image builder will work
+	dir string
+	// path to the ISO/RAW file from which a tarball will be created
+	imgPath string
+	// type of the image that will be used as base (either ISO or RAW)
+	imgType string
+	// podman client which to use for image build
+	podman Podman
```

Comment:
Nit: This could be simplified a lot by splitting up the Podman interface. What this type needs is an "image importer" having an `Import()` / `ImportImage()` method. Since the rest of the Podman functionality is not necessary, implementing the whole interface in unit tests will require mocking the other three functions which are never used.